src/brainglobe_napari/atlas_viewer_widget.py

```diff
@@ -0,0 +1,141 @@
+            if section == 0:
+                return "Atlas name"
+            elif section == 1:
+                return "Version"
+            else:
+                raise ValueError("Unexpected horizontal header value.")
+        else:
+            return super().headerData(section, orientation, role)
+
+
+class AtlasViewerWidget(QWidget):
+    """Widget to hold a selectable table view of atlases
+    and other widgets to visualise atlases in napari.
+
+    Internal state depends on currently selected row in the table view.
+    """
+
+    def __init__(self, napari_viewer: Viewer):
+        """Instantiates the atlas viewer widget
+        and sets up signal slot connections"""
+        super().__init__()
+
+        self._viewer = napari_viewer
+        self.setLayout(QVBoxLayout())
+
+        # setup atlas view
+        self.atlas_table_view = QTableView()
+        atlases = get_all_atlases_lastversions()
+        data = [[name, version] for name, version in atlases.items()]
+
+        self._model = AtlasTableModel(data)
+        self.atlas_table_view.setModel(self._model)
+        self.atlas_table_view.setEnabled(True)
+        self.atlas_table_view.verticalHeader().hide()
+
+        # select a single, entire row at a time.
+        self.atlas_table_view.setSelectionBehavior(
+            QAbstractItemView.SelectionBehavior.SelectRows
+        )
+        self.atlas_table_view.setSelectionMode(
+            QAbstractItemView.SelectionMode.SingleSelection
+        )
+
+        self._selected_atlas_row = None
+        self._selected_atlas_name = None
+
+        # set up add button
+        self.add_annotation_button = QPushButton()
+        self.add_annotation_button.setText("Add annotations layer")
+
+        def _on_add_annotations_clicked():
+            """Adds annotations as labels layer to the viewer."""
+            if self._selected_atlas_row is not None:
+                selected_atlas = BrainGlobeAtlas(self._selected_atlas_name)
+                napari_viewer.add_labels(
+                    selected_atlas.annotation, name=self._selected_atlas_name
+                )
+
+        self.add_annotation_button.clicked.connect(_on_add_annotations_clicked)
+
+        # set up atlas info display
+        self.atlas_info = QTextEdit(self)
+        self.atlas_info.setReadOnly(True)
+
+        # implement logic to update state when selection changes.
+        def _on_selection_changed():
+            """Updates the internally stored info about selected row."""
+            selected_index = (
+                self.atlas_table_view.selectionModel().currentIndex()
+            )
+            if selected_index.isValid():
+                self._selected_atlas_row = selected_index.row()
+                self._selected_atlas_name = self._model.data(
+                    self._model.index(self._selected_atlas_row, 0)
+                )
+                selected_atlas = BrainGlobeAtlas(self._selected_atlas_name)
```

Comment:
I think there should be some logic to make sure atlases (that can be 5GB+ currently) aren't downloaded just to look at some metadata.

Reply:
My feeling is that this would be a very large refactor to implement this in the atlas API "properly". For now, this widget could just have a `Download` button to make this step explicit to the user.

Reply:
I definitely agree with this, I will tackle as a separate issue #15